packages/@aws-cdk/core/lib/custom-resource-provider/export-reader-provider.ts

```diff
@@ -0,0 +1,119 @@
+import * as crypto from 'crypto';
+import * as path from 'path';
+import { Construct } from 'constructs';
+import { CfnResource } from '../cfn-resource';
+import { CustomResource } from '../custom-resource';
+import { Lazy } from '../lazy';
+import { Intrinsic } from '../private/intrinsic';
+import { Reference } from '../reference';
+import { Stack } from '../stack';
+import { CustomResourceProvider, CustomResourceProviderRuntime } from './custom-resource-provider';
+
+/**
+ * Properties for an ExportReader
+ */
+export interface ExportReaderProps {
+  /**
+   * The AWS region to read Stack exports from
+   *
+   * @default - the stack region
+   */
+  readonly region?: string;
+}
+
+/**
+ * Creates a custom resource that will return a list of stack exports from a given
+ * AWS region. The export can then be referenced by the export name.
+ *
+ *
+ * @example
+ * declare const app: App;
+ * const stack1 = new Stack(app, 'East1Stack', { env: { region: 'us-east-1' } });
+ * new CfnOutput(stack1, 'Output', { value: 'someValue', exportName: 'someName' });
+ *
+ * const stack2 = new Stack(app, 'East2Stack', { env: { region: 'us-east-2' } });
+ * const exportReader = new ExportReader(stack2, 'ExportReader', { region: 'us-east-1' });
+ * const anotherResource = new CfnResource(stack2, 'AnotherResource', {
+ *   Parameters: {
+ *     SomeParam: exportReader.importValue('someName'),
+ *   },
+ * });
+ *
+ * @internal - this is intentionally not exported from core
+ */
+export class ExportReader extends Construct {
+  private readonly resource: CustomResource;
+  private readonly _references: Reference[] = [];
+  constructor(scope: Construct, id: string, props: ExportReaderProps) {
+    super(scope, id);
+    const stack = Stack.of(this);
+    const region = props.region ?? stack.region;
+
+    const resourceType = 'Custom::CrossRegionExportReader';
+    const serviceToken = CustomResourceProvider.getOrCreate(this, resourceType, {
+      codeDirectory: path.join(__dirname, 'get-cfn-exports-handler'),
+      runtime: CustomResourceProviderRuntime.NODEJS_14_X,
+      policyStatements: [{
+        Effect: 'Allow',
+        Resource: '*',
+        Action: ['cloudformation:ListExports'],
+      }],
+    });
+
+    this.resource = new CustomResource(this, 'Default', {
+      resourceType: resourceType,
+      serviceToken,
+      properties: {
+        Region: region,
+        // This is used to determine when custom resource should be executed again.
+        //
+        // We want to lookup the resources whenever any of the references
+        // change. The only reliable way to tell whether we need to perform another lookup
+        // is to check if _any_ property of the referenced resource changes
+        RefreshToken: Lazy.string({
```

Comment:
I want some extra validation on this piece. I couldn't think of a better way of forcing the custom resource to run again, but maybe I am missing something.

Reply:
ahh yeah you are right. I like the idea of using SSM parameters, i'll look into that.

Reply:
ok i've updated it to use SSM parameters. 

Currently the SSM pararmeter gets a name like

`/cdk/exports/${stackName}/${exportName}`

I'm thinking though that I may have to also add accountId and region into there as well

`/cdk/exports/${accountId}/${region}/${stackName}/${exportName}`

Luckily the limit for names is is like 900 characters 😂